tripper/dataset/dataset.py

```diff
@@ -0,0 +1,736 @@
+                break
+        else:
+            raise ValueError(
+                f"dataset '{iri}' has no such distribution: {distribution}"
+            )
+    else:
+        distr = get(dct, "distribution")[0]
+
+    accessService = (
+        distr.accessService.get("endpointURL")
+        if "accessService" in distr
+        else None
+    )
+
+    # OTEAPI still puts the parse configurations into the dataresource
+    # instead of a in a separate parse strategy...
+    if parser:
+        if parser is True:
+            par = get(distr, "parser")[0]
+        elif isinstance(parser, str):
+            for par in get(distr, "parser"):
+                if par.get("@id") == parser:
+                    break
+            else:
+                raise ValueError(
+                    f"dataset '{iri}' has no such parser: {parser}"
+                )
+        configuration = par.get("configuration")
+    else:
+        configuration = None
+
+    dataresource = client.create_dataresource(
+        downloadUrl=distr.get("downloadURL"),
+        mediaType=distr.get("mediaType"),
+        accessUrl=distr.get("accessURL"),
+        accessService=accessService,
+        configuration=dict(configuration) if configuration else {},
+    )
+
+    statements = dct.get("statements", [])
+    statements.extend(dct.get("mappings", []))
+    if statements:
+        mapping = client.create_mapping(
+            mappingType="triples",
+            # The OTEAPI datamodels stupidly strict, requireing us
+            # to cast the data ts.namespaces and statements
+            prefixes={k: str(v) for k, v in ts.namespaces.items()},
+            triples=[tuple(t) for t in statements],
+        )
+
+    if parser:
+        pipeline = dataresource
+        if statements:
+            pipeline = pipeline >> mapping
+    elif generator:
+        if generator is True:
+            gen = get(distr, "generator")[0]
+        elif isinstance(generator, str):
+            for gen in get(distr, "generator"):
+                if gen.get("@id") == generator:
+                    break
+            else:
+                raise ValueError(
+                    f"dataset '{iri}' has no such generator: {generator}"
+                )
+
+        conf = gen.get("configuration")
+        if gen.generatorType == "application/vnd.dlite-generate":
+            conf.setdefault("datamodel", dct.get("datamodel"))
+
+        function = client.create_function(
+            functionType=gen.generatorType,
+            configuration=conf,
+        )
+        if statements:
+            pipeline = mapping >> function >> dataresource
+        else:
+            pipeline = function >> dataresource
+
+    return pipeline
+
+
+def list_dataset_iris(ts: Triplestore, **kwargs):
+    """Return a list of IRIs for all datasets matching a set of criterias
+    specified by `kwargs`.
+
+    Arguments:
+        ts: Triplestore to search.
+        kwargs: Match criterias.
```

Comment:
It is difficult for a new user to understand how this can be used. How about providing and example?

Reply:
Good point. Done